index.js

```diff
@@ -1,27 +1,81 @@
 var stylus  = require('stylus');
+var path    = require('path');
+var fs      = require('fs');
+
+var stylusUtils = require('stylus/lib/utils');
+var join = path.join;
+var dirname = path.dirname;
+var basename = path.basename;
+var relative = path.relative;
+
+var stylusLookupIndex = stylusUtils.lookupIndex;
+
+stylusUtils.lookupIndex = function(name, paths, filename){
+  var nodeModuleMatch = paths.find(includePath => includePath.endsWith(name));
+
+  var found = stylusLookupIndex(name, paths, filename);
+  
+  // check for node_modules/<name>
+  if (!found && nodeModuleMatch) {
+    found = stylusLookupIndex(nodeModuleMatch, paths, filename);
+  }
+  return found;
+
+  function lookupPackage(dir) {
+    var pkg = stylusUtils.lookup(join(dir, 'package.json'), paths, filename);
+    if (!pkg) {
+      return /\.styl$/i.test(dir) ? stylusUtils.lookupIndex(dir, paths, filename) : lookupPackage(dir + '.styl');
+    }
+    var main = require(relative(__dirname, pkg)).main;
+    if (main) {
+      found = stylusUtils.find(join(dir, main), paths, filename);
+    } else {
+      found = stylusUtils.lookupIndex(dir, paths, filename);
+    }
+    return found;
+  }
```

Comment:
This is never actually called and could be removed - the only call is a recursive call inside itself

Reply:
ah, good spot. Started from original fn implementation and didn't notice it was unused